### DISCUSSION TITLE
What is the right way to use customScrollParent?

### DISCUSSION THREAD

**Discussion by singpolyma** [petyosi/react-virtuoso#1104] [Q&A]
The example https://github.com/petyosi/react-virtuoso/blob/9a93e93dcbe5c02ad61dbd87d5e0c27251a583c6/examples/scroll-element-list-scroller.tsx#L5 shows using `ref={setSomeState}` but doing this results in an infinite re-render and react crashes. Using a `useRef` instead prevents the re-render, but means that customScrollParent is undefined on first mount and Virtuoso crashes.  Does anyone have a working example of using this?

**Comment by thomastvedt** [comment]
const scrollRef = useRef(null);
customScrollParent={scrollRef?.current || undefined}

Maybe this? 

Btw I'm looking for some examples on how to use customScrollPagent too if you find any ;)
